mmv1/provider/tflint.rb

```diff
@@ -1,15 +1,16 @@
-require 'provider/abstract_core'
-
 module Provider
   class TFLint < Provider::Terraform
     @@rule_names = []
-    @@api_products = {}
+    @@resource_url = {}
 
     def generate_resource(pwd, data, generate_code, generate_docs)
-      tf_product = (@config.legacy_name || data.product.name).underscore
-      @terraform_name = data.object.legacy_name || "google_#{tf_product}_#{data.object.name.underscore}"
+      @terraform_name = data.object.legacy_name || "google_#{full_resource_name(data)}"
 
-      @@api_products[@terraform_name] = data.product
+      begin
+        @@resource_url[@terraform_name] = URI.parse(data.product.base_url).host
+      rescue URI::InvalidURIError => exn
+        Google::LOGGER.warn "Cannot parse Api::Product#base_url: #{exn}"
```

Comment:
The following error may occur.

```
Cannot parse Api::Product#base_url: bad URI(is not URI?): "https://{{region}}-aiplatform.googleapis.com/v1/"
```

There's probably a proper way to prevent this, but I'll just ignore it for now.